From 449325143ed0d782775fa86b946d8a04b9225ccb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Wed, 9 Nov 2022 15:38:57 +0100
Subject: [PATCH] testsuite: skip tests that are currently failing on Alpine

To-Do: Investigate further why these tests fail in the first place.
---
 libraries/base/tests/IO/all.T          | 6 +++++-
 testsuite/tests/ghci/linking/all.T     | 2 ++
 testsuite/tests/ghci/linking/dyn/all.T | 2 ++
 testsuite/tests/rts/linker/all.T       | 2 +-
 4 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/libraries/base/tests/IO/all.T b/libraries/base/tests/IO/all.T
index 7cb99c0f7f..0dce4e8135 100644
--- a/libraries/base/tests/IO/all.T
+++ b/libraries/base/tests/IO/all.T
@@ -115,7 +115,11 @@ test('encoding001', [], compile_and_run, [''])
 
 test('encoding002', normal, compile_and_run, [''])
 test('encoding003', normal, compile_and_run, [''])
-test('encoding004', extra_files(['encoded-data/']), compile_and_run, [''])
+test('encoding004',
+    [when(platform('x86_64-alpine-linux'), skip),
+     when(platform('aarch64-alpine-linux'), skip),
+     extra_files(['encoded-data/'])],
+     compile_and_run, [''])
 test('encoding005', normal, compile_and_run, [''])
 
 test('environment001', [], makefile_test, ['environment001-test'])
diff --git a/testsuite/tests/ghci/linking/all.T b/testsuite/tests/ghci/linking/all.T
index 197500c039..f9e79dab6e 100644
--- a/testsuite/tests/ghci/linking/all.T
+++ b/testsuite/tests/ghci/linking/all.T
@@ -48,6 +48,8 @@ test('T3333',
 test('T11531',
      [extra_files(['T11531.hs', 'T11531.c', 'T11531.h']),
       unless(doing_ghci, skip),
+      when(platform('x86_64-alpine-linux'), skip),
+      when(platform('aarch64-alpine-linux'), skip),
       unless(opsys('linux'), skip),
       fragile(11531)],
      makefile_test, ['T11531'])
diff --git a/testsuite/tests/ghci/linking/dyn/all.T b/testsuite/tests/ghci/linking/dyn/all.T
index 9b05ed5fc3..52a25e637c 100644
--- a/testsuite/tests/ghci/linking/dyn/all.T
+++ b/testsuite/tests/ghci/linking/dyn/all.T
@@ -29,6 +29,8 @@ test('T10955dyn', [extra_files(['A.c', 'B.c', 'A.def', 'B.def'])],
 test('T10458',
      [extra_files(['A.c']),
       unless(doing_ghci, skip),
+      when(platform('x86_64-alpine-linux'), skip),
+      when(platform('aarch64-alpine-linux'), skip),
       pre_cmd('$MAKE -s --no-print-directory compile_libT10458'),
       extra_hc_opts('-L"$PWD/T10458dir" -lAS')],
      ghci_script, ['T10458.script'])
diff --git a/testsuite/tests/rts/linker/all.T b/testsuite/tests/rts/linker/all.T
index 4387bc2337..fbdad0ec59 100644
--- a/testsuite/tests/rts/linker/all.T
+++ b/testsuite/tests/rts/linker/all.T
@@ -89,7 +89,7 @@ test('linker_unload_native',
      [extra_files(['LinkerUnload.hs', 'Test.hs']),
       req_rts_linker,
       unless(have_dynamic(), skip),
-      when(opsys('darwin') or opsys('mingw32'), skip)],
+      when(opsys('darwin') or opsys('mingw32') or platform('x86_64-alpine-linux') or platform('aarch64-alpine-linux'), skip)],
      makefile_test, ['linker_unload_native'])
 
 ######################################
